### DISCUSSION TITLE
Help for a new developer setting up.

### DISCUSSION THREAD

**Discussion by dave-english** [scenerystack/community#37] [Q&A] [answered]
I'm trying to set up my development environment to work with SceneryStack library.  Starting by trying to load the "example-sim" on my desktop.  I've had Visual Studio Code installed on my PC for a while, and also have the node.js app installed, but I'm new to accessing files through github and only just installed the github app.  I've tried to launch the "example-sim" by copying the file location into my browser, which brings up a black screen and a long series of console errors.  Here is a typical error message:

example-sim_en.html:1 Access to script at 'file:///C:/Users/denglish/Documents/GitHub/chipper/js/load-unbuilt-strings.js' from origin 'null' has been blocked by CORS policy: Cross origin requests are only supported for protocol schemes: http, data, isolated-app, chrome-extension, chrome, https, chrome-untrusted.

Which I understand to mean that I can only access those files through a server protocol and not from my directory.  How do I set that up and launch the example-sim?"

**Comment by samreid** [comment]
I noticed the URL above starts with `file://` rather than `http://`. Could you try using a local http server, for example like: https://github.com/http-party/http-server so that you can run the simulation with something like `http://localhost/example-sim/example-sim_en.html`?

**Comment by dave-english** [comment]
> I'm trying to set up my development environment to work with SceneryStack library. Starting by trying to load the "example-sim" on my desktop. I've had Visual Studio Code installed on my PC for a while, and also have the node.js app installed, but I'm new to accessing files through github and only just installed the github app. I've tried to launch the "example-sim" by copying the file location into my browser, which brings up a black screen and a long series of console errors. Here is a typical error message:
> 
> example-sim_en.html:1 Access to script at 'file:///C:/Users/denglish/Documents/GitHub/chipper/js/load-unbuilt-strings.js' from origin 'null' has been blocked by CORS policy: Cross origin requests are only supported for protocol schemes: http, data, isolated-app, chrome-extension, chrome, https, chrome-untrusted.
> 
> Which I understand to mean that I can only access those files through a server protocol and not from my directory. How do I set that up and launch the example-sim?"

Thanks to Samreid for pointing me in the right direction.  I also found [this site](https://levelup.gitconnected.com/set-up-and-run-a-simple-node-server-project-38b403a3dc09) very useful for helping me understand server software in the most basic form.

I've tried running my project, now, through VSCode with the "Live Server" extension (Ritwick Dey), which gets me a little closer to launching the example sim.  Now the launched website shows a message "Adapted from PhET" and flashing load bar, with only two console errors.

> Live reload enabled.
> :5500/chipper/dist/js/example-sim/js/example-sim-main.js:1 
>         
>        Failed to load resource: the server responded with a status of 404 (Not Found)
> :5500/favicon.ico:1 
>         
>        Failed to load resource: the server responded with a status of 404 (Not Found)
> 

Which appear to suggest that my file structure is not set up correctly.  In my current configuration, the HTML file "example-sim_en" sits in a folder titled "example-sim".  The "chipper" folder (and others with names like "joist" and "scenery") sit in the same master projects folder as the "example-sim" folder.  What have I done wrong?

As a secondary follow up question, I recently acquired a Lenovo Flex 5 Chromebook, and plan to start using that machine for my development projects.  I've already installed Linux and VSCode on the Flex 5, but thinking of switching over to the cloud-based version of VSCode (to accomodate for the fact that Chromebook only has 250GB memory).  Anyone out there with experience working specifically in this coding environment?  Any potential pitfalls I should prepare for?

**Reply by zepumph** [reply, marked as answer]
Hello! Thanks for reaching out to us for assistance. We are glad and excited that you want to use SceneryStack for you work. For the first issue (the 404 on the main file), have you transpiled the source code yet? This is a precursor to running our unbuilt codebase locally. See https://github.com/phetsims/phet-info/blob/dfda01a439ce02dc77b887d484510ad17e8fcac2/doc/phet-development-overview.md#transpile-typescript for assistance. Furthermore, if you haven't seen that document before, I'd give it a read, as it may have more answers to future questions. Please forgive us as we get off the ground with SceneryStack documentation and communication. 

As for your VSCode questions, everyone on our project uses Jetbrains IDE products (Webstorm or Intellij). I don't have any experience with cloud based development, but I wish you luck!
